### PR TITLE
Add perl-Authen-SASL and perl-IO-Socket-SSL as requires to git to enable git send-email

### PR DESCRIPTION
**Pull request by aadhar-agarwal**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
- [Work Item](https://microsoft.visualstudio.com/OS/_sprints/taskboard/Mariner%20Platform/OS/2211?workitem=41520384) 
- Git send-mail fails due to missing perl dependencies (Perl-IO-Socket-SSL and Perl-Authen-SSL)
- This PR adds these missing dependencies to git.spec

###### Change Log  

- Added `Requires:       perl-Authen-SASL` and `Requires:       perl-IO-Socket-SSL` to git.spec
- Moved perl-Authen-SASL from SPECS-EXTENDED to SPECS
- Moved perl-GSSAPI from SPECS-EXTENDED to SPECS as perl-Authen-SASL depends on it (BuildRequires)

###### Does this affect the toolchain?  

**NO**

###### Test Methodology

- Buddy build (AMD64): https://dev.azure.com/mariner-org/mariner/_build/results?buildId=262198&view=results
- Buddy build (ARM64): https://dev.azure.com/mariner-org/mariner/_build/results?buildId=262199&view=results

In order to test this, I did the following:
- `docker run -it mcr.microsoft.com/cbl-mariner/base/core:2.0 /bin/bash`
- `mkdir testing`
- `cd testing`
- I copied over three 3 RPMS into the container: `git-2.33.4-2.cm2.x86_64.rpm`, `perl-GSSAPI-0.28-32.cm2.x86_64.rpm` and `perl-Authen-SASL-2.16-22.cm2.noarch.rpm`
- `tdnf install -y perl-GSSAPI-0.28-32.cm2.x86_64.rpm`
- `tdnf install -y  perl-Authen-SASL-2.16-22.cm2.noarch.rpm`
- `tdnf install -y git-2.33.4-2.cm2.x86_64.rpm`
- `git init`
- `echo "test" > test.txt`
- `git add test.txt`
- `git commit -m "first commit"`
- At this point, you should be prompted to fill out your name and email address (for git)
- I edited my gitconfig to look like this:
- image
- In order to use git send-email with gmail, you need to [set up](https://support.google.com/accounts/answer/6010255?hl=en) an app password 
- In order to authenticate with smtp.gmail.com, I got the cert from [here](https://support.google.com/a/answer/6180220?hl=en&ref_topic=2683828) with the hostname as `smtp.gmail.com` and added the cert to the end of /etc/ssl/certs/ca-bundle.crt
- `git send-email --format-patch -1 --to=aadagarwal@microsoft.com`

### CODE REVIEW

File: SPECS/git/git.spec [hunk 1]
```diff
@@ -168,6 +170,10 @@ fi
 %endif
 
 %changelog
+* Mon Oct 31 2022 Aadhar Agarwal <aadagarwal@microsoft.com> - 2.33.4-2
+- Add Requires on perl-Authen-SASL and perl-IO-Socket-SSL for git send-email
+- Bump version to 2.33.4-2
```
Comment (neha170) [inline SPECS/git/git.spec:175]: Please remove this line, as we're not changing the version.

**Review by neha170 — APPROVED** [review]

File: SPECS/perl-Authen-SASL/perl-Authen-SASL.spec [hunk 2]
```diff
@@ -2,7 +2,7 @@ Vendor:         Microsoft Corporation
 Distribution:   Mariner
 Name:           perl-Authen-SASL
 Version:        2.16
-Release:        21%{?dist}
+Release:        22%{?dist}
```
Comment (PawelWMS) [inline SPECS/perl-Authen-SASL/perl-Authen-SASL.spec:5]: Please lint the spec and perform a thorough spec clean-up (see: [instructions](https://dev.azure.com/mariner-org/mariner/_wiki/wikis/mariner.wiki/125/Introduction-to-SPEC-Files)). The core specs are required to follow more strict guidelines, so there's usually some additional work required when we migrate extended specs.

File: SPECS/perl-GSSAPI/perl-GSSAPI.spec [hunk 3]
```diff
@@ -8,7 +8,7 @@ Distribution:   Mariner
 
 Name:           perl-GSSAPI
 Version:        0.28
-Release:        31%{?dist}
+Release:        32%{?dist}
```
Comment (PawelWMS) [inline SPECS/perl-GSSAPI/perl-GSSAPI.spec:11]: Same clean-up request here.

**Review by PawelWMS — CHANGES_REQUESTED** [review]
Move looks fine as long as it build and the tests pass. We must clean-up the specs a bit, though, before we move them to core.

File: SPECS/perl-GSSAPI/perl-GSSAPI.spec (hunk 3 above)
Comment (PawelWMS) [inline SPECS/perl-GSSAPI/perl-GSSAPI.spec:11]: Please replace `%{?_with_testsuite:make test}` in the `%check` section with `make test` and verify the tests succeed. We always have network access during test builds, so Fedora's limitations don't apply.

File: SPECS/perl-Authen-SASL/perl-Authen-SASL.spec (hunk 2 above)
Comment (aadhar-agarwal) [reply SPECS/perl-Authen-SASL/perl-Authen-SASL.spec:5]: Sounds good, thank you for pointing this out :)

I will run the spec-cleaner on both the perl-Authen-SASL.spec and perl-GSSAPI.spec and verify that the tests pass

File: SPECS/perl-GSSAPI/perl-GSSAPI.spec (hunk 3 above)
Comment (aadhar-agarwal) [reply SPECS/perl-GSSAPI/perl-GSSAPI.spec:9]: Verified that the tests succeed

File: SPECS/perl-GSSAPI/perl-GSSAPI.spec [hunk 4]
```diff
@@ -1,34 +1,33 @@
-Vendor:         Microsoft Corporation
-Distribution:   Mariner
+Summary:        Perl extension providing access to the GSSAPIv2 library
 #
 # Rebuild option:
 #
 #   --with testsuite         - run the test suite
 #
```
Comment (PawelWMS) [inline SPECS/perl-GSSAPI/perl-GSSAPI.spec:6]: We should remove this as well, as that's not true for Mariner.
```suggestion
```

**Review by PawelWMS — APPROVED** [review]

**Pull request merged**